tests/lean/run/doNotation6.lean

```diff
@@ -12,7 +12,7 @@ if (← get) == 0 then
 modify (· - x)
 
 def f1 (x : Nat) : M Nat := do
-let v ←
+let v : Bool ←
```

Comment:
Is it clear to you why this annotation is necessary now?

Reply:
Actually I'm not sure anymore. The straightforward answer is that as the type annotation `Bool`, unrelated to anything in the expression, should indicate, the type of this variable is unconstrained, because the doElem before it doesn't return a value to the expression, so erroring with that would seem to make sense. On the other hand, the previous behavior was to not even elaborate the tail part of the expression (meaning for example that `v` would not be semantically highlighted as a variable) because that subterm ends up floating and not used in the final term output, in which case I don't think you would get the unbound type variable error, and testing it locally I am getting this behavior -- the type annotation is necessary, `v` is not highlighted and there are no errors.

I added this annotation in response to an error in CI: https://github.com/leanprover/lean4/actions/runs/6498883602/job/17651115144#step:11:2876 , but I'm personally fine with either behavior, it's a somewhat unimportant edge-case. LMK what behavior you want to see and I can try to implement it. For now I will remove the annotation and see if the error in CI persists.